# Faça a análise descritiva da variável SALARIO, para a base como um todo e considerando o Grau de Instrução

In [40]:
# importa biblioteca pandas
import pandas as pd

# local do arquivo
Caminho='/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'


In [41]:
df=pd.read_excel(Caminho)


In [42]:
df.head( )

ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [43]:
# gera estatísticas descritivas da coluna "idade" do DataFrame d
df.idade.describe()

count    426.000000
mean      34.622066
std        6.398190
min       20.000000
25%       30.000000
50%       34.000000
75%       40.000000
max       49.000000
Name: idade, dtype: float64

In [44]:
#calculando amplitude (determina a diferença entre o valor máximo e o valor mínimo de uma série de dados,
#  fornecendo uma medida simples da dispersão dos dados.)
ampl=df.idade.max()-df.idade.min()
ampl


29

In [45]:
#calculando a moda (identificar o valor que ocorre com maior frequência em uma série de dados, sendo uma 
# medida de tendência central que indica o valor mais comum.)
df.idade.mode()


0    35
Name: idade, dtype: int64

In [46]:
#calculando da variância (mede a dispersão dos dados em relação à média, indicando o quanto os valores de uma série 
# de dados variam em torno da média. Uma variância alta indica que os dados estão amplamente espalhados,
#  enquanto uma variância baixa indica que os dados estão mais próximos da média.)
df.idade.var()


40.93682960508146

In [47]:
# Calcula o valor mínimo da coluna 'idade' e armazena na variável Min
Min = df.idade.min()

# Calcula o valor máximo da coluna 'idade' e armazena na variável Max
Max = df.idade.max()

# Calcula a média (valor médio) da coluna 'idade' e armazena na variável Media
Media = df.idade.mean()

# Calcula a mediana (valor central) da coluna 'idade' e armazena na variável Mediana
Mediana = df.idade.median()

# Calcula a moda (valor mais frequente) da coluna 'idade' e armazena na variável Moda
Moda = df.idade.mode()

# Calcula a variância (medida de dispersão) da coluna 'idade' e armazena na variável Var
Var = df.idade.var()

# Calcula o desvio padrão (medida de dispersão) da coluna 'idade' e armazena na variável Desvpad
Desvpad = df.idade.std()


In [48]:
#calculado como a razão do desvio padrão pela média.
CV=df.idade.std() / df.idade.mean()
CV

0.18480091878825788

In [49]:
#Calculando descritivas por Região de Procedência
df.groupby(['reg_proc']) ['idade'].describe()


count      mean       std   min   25%   50%   75%   max
reg_proc                                                         
cap       143.0  36.06993  5.550854  22.0  33.0  35.0  39.5  49.0
int       135.0  33.77037  6.004899  20.0  29.5  33.0  37.5  49.0
out       148.0  34.00000  7.254368  20.0  28.0  34.0  40.0  46.0

In [50]:
# calculando amplitude por Região de Procedência

Ampl=df.groupby (['reg_proc']) ['idade'].max()-df.groupby (['reg_proc']) ['idade'].min()
Ampl


reg_proc
cap    27
int    29
out    26
Name: idade, dtype: int64

In [51]:
#calculando a variância por Região de Procedência
Var=df.groupby (['reg_proc']) ['idade'].var( )
Var



reg_proc
cap    30.811977
int    36.058817
out    52.625850
Name: idade, dtype: float64

In [52]:
#calculando o CV por Região de Procedência
CV= df.groupby (['reg_proc']) ['idade'].std( ) / df.groupby (['reg_proc']) ['idade'].mean( )
CV

reg_proc
cap    0.153891
int    0.177816
out    0.213364
Name: idade, dtype: float64

In [53]:
#calculando a moda
Moda=df.groupby(['reg_proc'])['idade'].agg(pd.Series.mode)
Moda

reg_proc
cap          35
int    [30, 34]
out    [33, 41]
Name: idade, dtype: object

# Análise Descritiva da Variável SALARIO para a Base como um Todo


In [54]:
# Importar as bibliotecas necessárias e carregar o dataset:
import pandas as pd

# Local do arquivo
Caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'
df = pd.read_excel(Caminho)

In [55]:
# Descrever a variável SALARIO:
df.salario.describe()

count    426.000000
mean      11.080657
std        4.215989
min        4.000000
25%        7.650000
50%       10.400000
75%       13.900000
max       23.900000
Name: salario, dtype: float64

In [56]:
# Calcular a amplitude
ampl = df['salario'].max() - df['salario'].min()
ampl

19.9

In [57]:
# outra forma de Calcular a amplitude
ampl=df.salario.max()-df.salario.min()
ampl

19.9

In [58]:
# Calcular a moda:
df.salario.mode()


0     9.8
1    13.6
Name: salario, dtype: float64

In [59]:
#calculando da variância
df.salario.var()

17.774563802264566

In [60]:
# Calcular o coeficiente de variação (CV):
cv = df['salario'].std() / df['salario'].mean()
cv

0.3804818577389044

### Análise Descritiva da Variável SALARIO Considerando o Grau de Instrução

In [61]:
#Calculando descritivas por grau_instrucao
df.groupby(["grau_instrucao"]) ['salario'].describe()

count       mean       std   min      25%     50%     75%  \
grau_instrucao                                                              
f               120.0   7.922667  2.802770  4.00   6.2475   7.365   8.995   
m               203.0  11.208522  3.785457  4.43   8.2600  10.500  13.850   
s               103.0  14.507864  3.590439  7.60  11.3500  14.550  17.260   

                 max  
grau_instrucao        
f               14.7  
m               23.9  
s               23.5

In [62]:
#calculando a moda
Moda=df.groupby(['grau_instrucao'])['salario'].agg(lambda x: pd.Series.mode(x)[0]) # Using a lambda function to extract the first mode
Moda

grau_instrucao
f     6.26
m    13.60
s    18.75
Name: salario, dtype: float64

In [63]:
Ampl=df.groupby (['grau_instrucao']) ['salario'].max()-df.groupby (['grau_instrucao']) ['salario'].min()
Ampl

grau_instrucao
f    10.70
m    19.47
s    15.90
Name: salario, dtype: float64

In [64]:
#calculando a variância por salario
Var=df.groupby (['grau_instrucao']) ['salario'].var( )
Var


grau_instrucao
f     7.855517
m    14.329684
s    12.891250
Name: salario, dtype: float64

In [65]:
#calculando o CV por salario
CV= df.groupby (['grau_instrucao']) ['salario'].std( ) / df.groupby (['grau_instrucao']) ['salario'].mean( )
CV

grau_instrucao
f    0.353766
m    0.337730
s    0.247482
Name: salario, dtype: float64

In [66]:
#calculando a moda
Moda=df.groupby(['grau_instrucao'])['salario'].agg(lambda x: pd.Series.mode(x)[0]) # Using a lambda function to extract the first mode
Moda

grau_instrucao
f     6.26
m    13.60
s    18.75
Name: salario, dtype: float64

# ADICIONAL Outra forma de Apresentar uma análise descritiva da variável SALARIO tanto para a base como um todo quanto considerando o grau_instrucao.

### 1. Importação das Bibliotecas Necessárias:

In [67]:
import pandas as pd

### 2. Carregamento do Dataset:

In [68]:
Caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'
df = pd.read_excel(Caminho)

### 3. Estatísticas Descritivas para a Variável SALARIO para a Base como um Todo:

In [69]:
estatisticas_salario = df['salario'].describe()
print("Estatísticas descritivas para SALARIO (base como um todo):")
print(estatisticas_salario)

Estatísticas descritivas para SALARIO (base como um todo):
count    426.000000
mean      11.080657
std        4.215989
min        4.000000
25%        7.650000
50%       10.400000
75%       13.900000
max       23.900000
Name: salario, dtype: float64


### 4. Cálculo da Amplitude:

In [70]:
amplitude_salario = df['salario'].max() - df['salario'].min()
print("\nAmplitude de SALARIO (base como um todo):", amplitude_salario)


Amplitude de SALARIO (base como um todo): 19.9


### 5. Cálculo da Moda:

In [71]:
moda_salario = df['salario'].mode()[0]
print("\nModa de SALARIO (base como um todo):", moda_salario)


Moda de SALARIO (base como um todo): 9.8


### 6. Cálculo da Variância:

In [72]:
variancia_salario = df['salario'].var()
print("\nVariância de SALARIO (base como um todo):", variancia_salario)


Variância de SALARIO (base como um todo): 17.774563802264566


### 7. Cálculo do Coeficiente de Variação (CV):

In [73]:
cv_salario = df['salario'].std() / df['salario'].mean()
print("\nCoeficiente de variação de SALARIO (base como um todo):", cv_salario)


Coeficiente de variação de SALARIO (base como um todo): 0.3804818577389044


### 8. Estatísticas Descritivas para a Variável SALARIO Considerando o Grau de Instrução:

In [74]:
estatisticas_salario_grau_instrucao = df.groupby('grau_instrucao')['salario'].describe()
print("\nEstatísticas descritivas para SALARIO considerando grau_instrucao:")
print(estatisticas_salario_grau_instrucao)


Estatísticas descritivas para SALARIO considerando grau_instrucao:
                count       mean       std   min      25%     50%     75%  \
grau_instrucao                                                              
f               120.0   7.922667  2.802770  4.00   6.2475   7.365   8.995   
m               203.0  11.208522  3.785457  4.43   8.2600  10.500  13.850   
s               103.0  14.507864  3.590439  7.60  11.3500  14.550  17.260   

                 max  
grau_instrucao        
f               14.7  
m               23.9  
s               23.5  


### 9. Cálculo da Amplitude por Grau de Instrução:

In [75]:
amplitude_salario_grau_instrucao = df.groupby('grau_instrucao')['salario'].max() - df.groupby('grau_instrucao')['salario'].min()
print("\nAmplitude de SALARIO por grau_instrucao:")
print(amplitude_salario_grau_instrucao)


Amplitude de SALARIO por grau_instrucao:
grau_instrucao
f    10.70
m    19.47
s    15.90
Name: salario, dtype: float64


### 10. Cálculo da Moda por Grau de Instrução:

In [76]:
moda_salario_grau_instrucao = df.groupby('grau_instrucao')['salario'].agg(lambda x: pd.Series.mode(x)[0])
print("\nModa de SALARIO por grau_instrucao:")
print(moda_salario_grau_instrucao)


Moda de SALARIO por grau_instrucao:
grau_instrucao
f     6.26
m    13.60
s    18.75
Name: salario, dtype: float64


### 11. Cálculo da Variância por Grau de Instrução:

In [77]:
variancia_salario_grau_instrucao = df.groupby('grau_instrucao')['salario'].var()
print("\nVariância de SALARIO por grau_instrucao:")
print(variancia_salario_grau_instrucao)


Variância de SALARIO por grau_instrucao:
grau_instrucao
f     7.855517
m    14.329684
s    12.891250
Name: salario, dtype: float64


### 12. Cálculo do Coeficiente de Variação (CV) por Grau de Instrução:

In [78]:
cv_salario_grau_instrucao = df.groupby('grau_instrucao')['salario'].std() / df.groupby('grau_instrucao')['salario'].mean()
print("\nCoeficiente de variação de SALARIO por grau_instrucao:")
print(cv_salario_grau_instrucao)


Coeficiente de variação de SALARIO por grau_instrucao:
grau_instrucao
f    0.353766
m    0.337730
s    0.247482
Name: salario, dtype: float64


## Interpretaçao dos Dados da Análise Descritiva da Variável SALARIO

### 1- Para a Base como um Todo:

. *Estatísticas Descritivas:*

| Estatística | Valor |
|-------------|-------|
| count       | 426.000000 |
| mean        | 11.080657 |
| std         | 4.215989 |
| min         | 4.000000 |
| 25%         | 7.650000 |
| 50%         | 10.400000 |
| 75%         | 13.900000 |
| max         | 23.900000 |
| Name        | salario |
| dtype       | float64 |

**Contagem (count):** 426  
**Média (mean):** 11.08  
**Desvio Padrão (std):** 4.22  
**Mínimo (min):** 4.00  
**1º Quartil (25%):** 7.65  
**Mediana (50%):** 10.40  
**3º Quartil (75%):** 13.90  
**Máximo (max):** 23.90  


. *Amplitude:*
19.9

. *Variância:*
17.774563802264566

. *Coeficiente de Variação (CV):*
0.3804818577389044

### 2- Considerando o Grau de Instrução:

. *Estatísticas Descritivas:*

                            count       mean       std   min      25%     50%     75%    max
            grau_instrucao                                                             
            f               120.0   7.922667  2.802770  4.00   6.2475   7.365   8.995   14.7
            m               203.0  11.208522  3.785457  4.43   8.2600  10.500  13.850   23.9
            s               103.0  14.507864  3.590439  7.60  11.3500  14.550  17.260   23.5

## Conclusão Análise Descritiva da Variável SALARIO

A análise descritiva da variável SALARIO mostra que, em geral, os salários variam significativamente com o grau de instrução. A média e a variância dos salários aumentam com o nível de instrução, indicando que pessoas com maior grau de instrução tendem a ter salários mais altos e uma maior dispersão nos salários.

### 1. Para a Base como um Todo:

#### Estatísticas Descritivas:

| Estatística | Valor       |
|-------------|-------------|
| count       | 426.000000  |
| mean        | 11.080657   |
| std         | 4.215989    |
| min         | 4.000000    |
| 25%         | 7.650000    |
| 50%         | 10.400000   |
| 75%         | 13.900000   |
| max         | 23.900000   |

#### Amplitude:
19.9

#### Variância:
17.774563802264566

#### Coeficiente de Variação (CV):
0.3804818577389044

### 2. Considerando o Grau de Instrução:

#### Estatísticas Descritivas:

| Grau de Instrução | count | mean      | std      | min  | 25%    | 50%    | 75%    | max  |
|-------------------|-------|-----------|----------|------|--------|--------|--------|------|
| Fundamental (f)   | 120.0 | 7.922667  | 2.802770 | 4.00 | 6.2475 | 7.365  | 8.995  | 14.7 |
| Médio (m)         | 203.0 | 11.208522 | 3.785457 | 4.43 | 8.2600 | 10.500 | 13.850 | 23.9 |
| Superior (s)      | 103.0 | 14.507864 | 3.590439 | 7.60 | 11.3500| 14.550 | 17.260 | 23.5 |

#### Amplitude por Grau de Instrução:

| Grau de Instrução | Amplitude |
|-------------------|-----------|
| Fundamental (f)   | 10.70     |
| Médio (m)         | 19.47     |
| Superior (s)      | 15.90     |

#### Moda por Grau de Instrução:

| Grau de Instrução | Moda  |
|-------------------|-------|
| Fundamental (f)   | 6.26  |
| Médio (m)         | 13.60 |
| Superior (s)      | 18.75 |

#### Variância por Grau de Instrução:

| Grau de Instrução | Variância |
|-------------------|-----------|
| Fundamental (f)   | 7.855517  |
| Médio (m)         | 14.329684 |
| Superior (s)      | 12.891250 |

#### Coeficiente de Variação (CV) por Grau de Instrução:

| Grau de Instrução | CV       |
|-------------------|----------|
| Fundamental (f)   | 0.353766 |
| Médio (m)         | 0.337730 |
| Superior (s)      | 0.247482 |